geoh5py/shared/entity.py

```diff
@@ -237,10 +233,23 @@ def metadata(self) -> dict | None:
     @metadata.setter
     def metadata(self, value: dict | None):
         if value is not None:
-            assert isinstance(
-                value, (dict, str)
-            ), f"Input metadata must be of type {dict} or None"
-        self._metadata = value
+            if not isinstance(value, dict):
```

Comment:
You can remove one level of indentation if you treat each case individually. It will read better too

```suggestion
        if isinstance(value, dict):
            do the dict operations
            
        elif value is None:
            self._metadata = None
            
        else:
            raise TypeError(
                    "Input metadata must be of type dict or None"
                    f" find '{type(value)}'."
                )
        
        self.workspace.update_attribute(self, "metadata")
        
```